### PR TITLE
[DB] Add database schema

### PR DESCRIPTION
**Pull request by Natata08**
# [DB] Add database schema and updating README

---

## Description

This PR adds the initial database schema for the application's PostgreSQL database. The schema defines the core tables and relationships needed for managing job applications, companies, user data, and related entities.
Also, this PR updated README, adding all information for setting up your local database for Unix-based systems and Windows. 

---

## Changes Made

Added `api/db/schema.sql` file with the following key components:
- Core Tables:
   - user - User account information
   - company - Company details
   - application - Job application tracking
   - company_contact - Company contact information
   - application_note - Notes for applications
   - document - Document management
   - interview - Interview scheduling
   - notification - User notifications

- Additional Features:
   - Custom ENUM types for application status and document types
   - Automatic timestamp updates via triggers
   - Appropriate foreign key constraints
   - Indexes for performance optimization
   - Computed column for active applications

### CODE REVIEW

**Review by pankaj28843 — APPROVED** [review]

File: api/db/schema.sql [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+);
+
+CREATE TABLE "company_contact" (
+    contact_id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+    company_id INT NOT NULL REFERENCES "company" (company_id) ON DELETE CASCADE,
+    application_id INT NOT NULL REFERENCES "application" (application_id) ON DELETE CASCADE,
+    name VARCHAR(100) NOT NULL,
+    phone VARCHAR(100),
+    email VARCHAR(255),
+    ROLE VARCHAR(100),
```
Comment (olhasashchuk) [inline api/db/schema.sql:27]: Why ROLE - uppercase?

File: api/db/schema.sql [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    applied_date DATE,
+    deadline_date DATE,
+    created_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    updated_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP
+);
+
+CREATE TABLE "application_note" (
+    note_id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+    application_id INT NOT NULL REFERENCES "application" (application_id) ON DELETE CASCADE,
+    CONTENT TEXT NOT NULL,
```
Comment (olhasashchuk) [inline api/db/schema.sql:67]: Also here Upper case

File: api/db/schema.sql [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    user_id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+    password_hash VARCHAR(255) NOT NULL,
+    first_name VARCHAR(100) NOT NULL,
+    last_name VARCHAR(100) NOT NULL,
+    email VARCHAR(255) UNIQUE NOT NULL,
+    created_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    updated_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP
+);
+
+CREATE TABLE "company" (
```
Comment (pankaj28843) [inline api/db/schema.sql:11]: add readme for using this sql script and make other changes like discussed on Slack

**Review by pankaj28843 — CHANGES_REQUESTED** [review]

File: api/db/schema.sql (hunk 1 above)
Comment (Natata08) [reply api/db/schema.sql:27]: Prettier makes it uppercase, but it doesn't influence on table column.

**Review by pankaj28843 — DISMISSED** [review]

**Review by olhasashchuk — APPROVED** [review]

**Review by pankaj28843 — APPROVED** [review]

**Review by myselfankitaa — APPROVED** [review]

**Pull request merged**